### ISSUE TITLE
support variable substitution in settings

### ISSUE THREAD

**Issue by jason-ha** [prettier/prettier-vscode#1351]
Working with a repository that has prettier installed via npm outside of workspace folder, but in a relatively named location. For example if workspace is `.../repo/`, then prettier may be found in `.../repo.bin/tools/node_modules/prettier`.

VSCode variables do not appear to be supported as specifying
```
  	"prettier.prettierPath": "${workspaceFolder}.bin/tools/node_modules/prettier",
```
results in failure to load prettier module with message similar to:
```
["ERROR" - 2:27:31 PM] Error clearing module cache.
["ERROR" - 2:27:31 PM] Cannot find module '...\repo\${workspaceFolder}.bin\tools\node_modules\prettier'
```

Would like `${workspaceFolder}` and ideally all `${command:foo}` substitutions to be supported.

Related issue is that `.prettierrc.js` must also be located as sibling to `node_modules` if it references any other packages (and requires similar `prettier.configPath` specification). Leaving the `.prettierrc.js` in repo root will result in failed load of referenced packages.
[edit - below original statement is incorrect - ignore]
prettier supports the latter when cwd is `.../repo.bin/tools`. So alternative may be to allow `workingDirectory` specification.

**Comment by ntotten** [comment]
Is there any other extension that has done this? This seems more like a feature VSCode needs to implement rather than us creating our own way to do this.

**Comment by jason-ha** [comment]
I was under the impression that `gtest-adapter` had support for `${workspaceFolder}` substitution, but the related substitution is not in settings, but in `launch.json` or `settings "launch"` section. While the page you listed suggests some substitutions are supported `${workspaceFolder}` is not documented and I guess must be given special treatment when found in `settings "launch"` values. I agree that per extension workaround is not the way to go.
Our workaround is to generate `settings.json` with a repo bootstrapping script.

**Comment by jason-ha** [comment]
I looked for VS Code issue on this or similar and most relevant issue I found was [#97590 add support for configuration variables (config) settings.json' values](https://github.com/microsoft/vscode/issues/97590). From there I found that at least one extension is adding their own support, though the implementation is flawed and that backups up idea that extensions don't do this all on their own. Also found VS Code internal support, but I didn't see that it is exposed.
